### ISSUE TITLE
Add a cargo command for kythe [phab:D1010]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1346]
_Created by **djrenren** at 2016-08-13 00:48:57_: 

This adds an indexer driver for cargo so that one could write:

- `cargo kythe` for a top-level index

- `cargo kythe full-index` for a deep index including dependencies

- `cargo kythe web` to run the web UI

The utility requires a KYTHE_DIR environment variable to set and complains if it is not present.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-08-15 16:55:09_:

Please add a README with instructions on how to build and run the tool.

Bazel complains when I tried to do the obvious thing:
```
bazel build //kythe/rust/cargo-kythe
..
WARNING: kythe-trunk/kythe/rust/cargo-kythe/BUILD:5:1: target 'cargo-kythe' is both a rule and a file; please choose another name for the rule.
INFO: Found 1 target...
ERROR: kythe-trunk/kythe/rust/cargo-kythe/BUILD:5:1: Executing genrule //kythe/rust/cargo-kythe:cargo-kythe failed: bash failed: error executing command /bin/bash -c ... (remaining 1 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 101.
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling rustc-serialize v0.3.19
   Compiling indexer v0.1.0 (file://kythe-trunk/kythe/rust/indexer)
   Compiling cargo-kythe v0.1.0 (file://kythe-trunk/kythe/rust/cargo-kythe)
error: failed to run custom build command for `cargo-kythe v0.1.0 (file://kythe-trunk/kythe/rust/cargo-kythe)`
process didn't exit successfully: `kythe-trunk/kythe/rust/cargo-kythe/target/debug/build/cargo-kythe-b4d6f68a7c42e6f5/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', ../src/libcore/result.rs:788
note: Run with `RUST_BACKTRACE=1` for a backtrace.

Target //kythe/rust/cargo-kythe:cargo-kythe failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 25.990s, Critical Path: 20.42s
```

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-08-15 16:56:33_:

nit: whitespace

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-08-15 17:43:43_:

  - Add expect messages to build.rs

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-08-15 22:54:04_:

  - Prep for publishing. Remove from bazel build path

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-08-15 23:02:41_:

nit: `cargo-kythe`

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-08-15 23:02:41_:

nit: use `cargo` (e.g., the monospaced command name) or Cargo (the proper name of the tool).

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-08-15 23:02:41_:

Where are the index data written and how can that be configured?

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-08-15 23:02:41_:

nit: you probably want "dependencies" here, since "dependents" goes the other way.

**Comment by kythe** [comment]
_**djrenren** wrote at 2016-08-16 17:59:47_:

  - Fix README nits

**Comment by kythe** [comment]
_**zarko** wrote at 2016-08-16 19:47:39_:

Before you publish this to crates.io (or elsewhere), we need to package a new Kythe binary release that includes #1339.

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2016-08-16 19:47:39_

**Comment by kythe** [comment]
_**djrenren** closed via commit 0a4e1e8021309abb8f377dd09bf12687711b2f10 at 2016-08-16 21:00:09_
